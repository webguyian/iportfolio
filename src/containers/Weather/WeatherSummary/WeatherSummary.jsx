import React from 'react';
import PropTypes from 'prop-types';

import Text from 'components/Text/Text';

import {
  getPercentage,
  getPressure,
  getTime,
  getWindDirection,
  formatTemp
} from 'modules/weather/helpers';

const WeatherSummary = props => {
  const { currentHour, today } = props;
  const baseClass = 'weather-summary';

  return (
    <div className={baseClass}>
      <div className={`${baseClass}-description`}>
        <Text element="p">Today: {today.summary}</Text>
      </div>
      <div className={`${baseClass}-table`}>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Sunrise
            </Text>
            <Text>{getTime(today.sunriseTime)}</Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Sunset
            </Text>
            <Text>{getTime(today.sunsetTime)}</Text>
          </div>
        </div>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Chance of rain
            </Text>
            <Text>{getPercentage(today.precipProbability)}</Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Humidity
            </Text>
            <Text>{getPercentage(today.humidity)}</Text>
          </div>
        </div>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Wind
            </Text>
            <Text>
              {getWindDirection(today.windBearing)}{' '}
              {Math.round(today.windSpeed)} mph
            </Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Feels like
            </Text>
            <Text>{formatTemp(currentHour.apparentTemperature)}</Text>
          </div>
        </div>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Precipitation
            </Text>
            <Text>{Math.round(today.precipIntensity)} in</Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Pressure
            </Text>
            <Text>{getPressure(today.pressure)} inHg</Text>
          </div>
        </div>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Visibility
            </Text>
            <Text>{today.visibility} mi</Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              UX index
            </Text>
            <Text>{today.uvIndex}</Text>
          </div>
        </div>
        <div className={`${baseClass}-table-row`}>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Dew Point
            </Text>
            <Text>{formatTemp(today.dewPoint)}</Text>
          </div>
          <div className={`${baseClass}-table-cell`}>
            <Text className="ui-text--muted" element="h3">
              Ozone
            </Text>
            <Text>{Math.round(today.ozone)} DU</Text>
          </div>
        </div>
      </div>
    </div>
  );
};

WeatherSummary.propTypes = {
  currentHour: PropTypes.object.isRequired,
  today: PropTypes.object.isRequired
};

export default WeatherSummary;
