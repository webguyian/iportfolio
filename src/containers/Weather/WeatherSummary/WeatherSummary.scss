.weather-summary {
  &-description {
    margin-bottom: 8px;
    padding: $gutter 8px;
    border-top: $border;
    border-bottom: $border;

    > .ui-text--body {
      margin: 0;
      text-align: left;
    }
  }

  &-table {
    &-row {
      display: flex;
      flex-flow: row nowrap;
      text-align: left;
    }

    &-cell {
      flex: 0 1 50%;
      padding: 8px;
    }

    .ui-text--muted {
      margin: 0;
      color: rgba($white, 0.5); // stylelint-disable-line
      font-size: $font-size-s;
      text-transform: uppercase;
    }
  }
}
