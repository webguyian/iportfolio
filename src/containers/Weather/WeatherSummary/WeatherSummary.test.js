import React from 'react';
import { create } from 'react-test-renderer';

import WeatherSummary from './WeatherSummary';

describe('<WeatherSummary />', () => {
  const props = {
    currentHour: {
      time: 1583676000,
      summary: 'Clear',
      icon: 'clear-day',
      temperature: 41.17,
      apparentTemperature: 41.17
    },
    today: {
      time: 1583643600,
      summary: 'Clear throughout the day.',
      icon: 'clear-day',
      sunriseTime: 1583666580,
      sunsetTime: 1583708460,
      moonPhase: 0.47,
      precipIntensity: 0.0004,
      precipIntensityMax: 0.002,
      precipIntensityMaxTime: 1583703480,
      precipProbability: 0.03,
      precipType: 'snow',
      precipAccumulation: 0.04,
      temperatureHigh: 58.95,
      temperatureHighTime: 1583698500,
      temperatureLow: 38.77,
      temperatureLowTime: 1583752380,
      apparentTemperatureHigh: 58.45,
      apparentTemperatureHighTime: 1583698500,
      apparentTemperatureLow: 33.74,
      apparentTemperatureLowTime: 1583752260,
      dewPoint: 24.59,
      humidity: 0.53,
      pressure: 1031.5,
      windSpeed: 5.68,
      windGust: 23.83,
      windGustTime: 1583726400,
      windBearing: 237,
      cloudCover: 0.01,
      uvIndex: 5,
      uvIndexTime: 1583687400,
      visibility: 10,
      ozone: 341.9,
      temperatureMin: 26.36,
      temperatureMinTime: 1583667300,
      temperatureMax: 58.95,
      temperatureMaxTime: 1583698500,
      apparentTemperatureMin: 26.85,
      apparentTemperatureMinTime: 1583667300,
      apparentTemperatureMax: 58.45,
      apparentTemperatureMaxTime: 1583698500
    }
  };

  it('renders correctly', () => {
    const component = create(<WeatherSummary {...props} />);

    expect(component).toMatchSnapshot();
  });
});
