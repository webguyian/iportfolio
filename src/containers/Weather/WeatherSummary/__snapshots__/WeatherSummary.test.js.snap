// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherSummary /> renders correctly 1`] = `
<div
  className="weather-summary"
>
  <div
    className="weather-summary-description"
  >
    <p
      className="ui-text ui-text--body"
    >
      Today: 
      Clear throughout the day.
    </p>
  </div>
  <div
    className="weather-summary-table"
  >
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Sunrise
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          11:11AM
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Sunset
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          11:11AM
        </span>
      </div>
    </div>
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Chance of rain
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          3%
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Humidity
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          53%
        </span>
      </div>
    </div>
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Wind
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          WSW
           
          6
           mph
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Feels like
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          41°
        </span>
      </div>
    </div>
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Precipitation
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          0
           in
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Pressure
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          30.46
           inHg
        </span>
      </div>
    </div>
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Visibility
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          10
           mi
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          UX index
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          5
        </span>
      </div>
    </div>
    <div
      className="weather-summary-table-row"
    >
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Dew Point
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          25°
        </span>
      </div>
      <div
        className="weather-summary-table-cell"
      >
        <h3
          className="ui-text ui-text--body ui-text--muted"
        >
          Ozone
        </h3>
        <span
          className="ui-text ui-text--body"
        >
          342
           DU
        </span>
      </div>
    </div>
  </div>
</div>
`;
